contracts/dev/DevAdmin.sol

```diff
@@ -182,4 +205,199 @@ contract DevAdmin is AccessControlInternal {
+        address sdIdProxyAddress = SyntheticDeviceStorage
+            .getStorage()
+            .idProxyAddress;
+
+        uint256 tokenId;
+        address owner;
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            tokenId = tokenIds[i];
+
+            if (!INFT(vehicleIdProxyAddress).exists(tokenId))
+                revert InvalidNode(vehicleIdProxyAddress, tokenId);
+            // Check AD pairing
+            if (ms.links[vehicleIdProxyAddress][tokenId] != 0)
+                revert VehiclePaired(tokenId);
+            // Check SD pairing
+            if (
+                ms.nodeLinks[vehicleIdProxyAddress][sdIdProxyAddress][
+                    tokenId
+                ] != 0
+            ) revert VehiclePaired(tokenId);
+
+            owner = INFT(vehicleIdProxyAddress).ownerOf(tokenId);
+
+            delete ns.nodes[vehicleIdProxyAddress][tokenId].parentNode;
+
+            emit VehicleNodeBurnedDevAdmin(tokenId, owner);
+
+            INFT(vehicleIdProxyAddress).burn(tokenId);
+
+            _resetVehicleInfos(tokenId);
+        }
+    }
+
+    /**
+     * @notice Admin function to burn a list of vehicles and reset all its attributes and links
+     * @dev Caller must have the admin role
+     * @dev This contract has the BURNER_ROLE in the VehicleId
+     * @param tokenIds List of vehicle node ids
+     */
+    function adminBurnVehiclesAndDeletePairings(uint256[] calldata tokenIds)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+        SyntheticDeviceStorage.Storage storage sds = SyntheticDeviceStorage
+            .getStorage();
+
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        address adIdProxyAddress = AftermarketDeviceStorage
+            .getStorage()
+            .idProxyAddress;
+        address sdIdProxyAddress = SyntheticDeviceStorage
+            .getStorage()
+            .idProxyAddress;
+
+        uint256 tokenId;
+        uint256 pairedNode;
+        address owner;
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            tokenId = tokenIds[i];
+
+            if (!INFT(vehicleIdProxyAddress).exists(tokenId))
+                revert InvalidNode(vehicleIdProxyAddress, tokenId);
+
+            owner = INFT(vehicleIdProxyAddress).ownerOf(tokenId);
+
+            // Check AD pairing
+            pairedNode = ms.links[vehicleIdProxyAddress][tokenId];
+            if (pairedNode != 0) {
+                delete ms.links[vehicleIdProxyAddress][tokenId];
+                delete ms.links[adIdProxyAddress][pairedNode];
+
+                emit AftermarketDeviceUnpairedDevAdmin(
+                    pairedNode,
+                    tokenId,
+                    owner
+                );
+                // Check SD pairing
+            } else if (
```

Comment:
Why is this `else if`? Could one not have both?

Reply:
Not supposed to I believe. But in fact we allow that

Reply:
No, I don't get it—plenty of our vehicles have Smartcar and AutoPi attached.

Reply:
My bad